tests/lexer/tstrnumlits.nim

```diff
@@ -0,0 +1,91 @@
+discard """
+  output: "[[1]]neg[[1]][[1]][[12345.67890]][[12345.67890]][[1]]times[[1]][[1]]plusneg[[1]][[1]][[22.40]][[0x5a3a]]neg[[38383839292839283928392839283928392839283.928493849385935898243e-50000]]"
+"""
+# Test tkStrNumLit
+
+type
+  S = distinct string
+
+proc wrap(number: string): S =
+  result = ("[[" & number & "]]").S
+
+template twrap(number: string): untyped =
+  wrap(number)
+
+proc extraContext(): S =
+  22.40wrap
+
+# if a user is writing a numberic library, they might write proc similar to:
+
+proc `*`(left, right: S): S =
+  result = (left.string & "times" & right.string).S
+
+proc `+`(left, right: S): S =
+  result = (left.string & "plus" & right.string).S
+
+proc `-`(number: S): S =
+  result = ("neg" & number.string).S
+
+proc `$`(number: S): string =
+  result = number.string
+
+const
+  likeInteger    = 1wrap
```

Comment:
where is the quote? it should be `12344'suffix`, not `12344suffix`

Reply:
@timotheecour I think it was made so it's consistent with normal Nim numbers - "1i32" is valid

Reply:
good point, but it causes ambiguities, eg: `echo 12e32` : is it "12".e32 or the FP 12.0*10^32 ?

i don't see the point of supporting the non-quoted version

Reply:
Can't we specify that proper `e32` and `x32` etc always take priority? Couldn't we also require quotes specifically for hex or binary or engineering notation? So `1wrap` is fine, but `1e32wrap` must be `1e32'wrap`.